src/main/java/org/jenkinsci/plugins/github/pullrequest/builders/GitHubPRStatusBuilder.java

```diff
@@ -52,41 +56,44 @@ public GitHubPRMessage getStatusMessage() {
     }
 
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
-            throws InterruptedException, IOException {
-        GitHubPRTrigger trigger = build.getProject().getTrigger(GitHubPRTrigger.class);
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher,
+                        @Nonnull TaskListener listener) throws InterruptedException, IOException {
+        // No triggers in Run class, but we need it
+        final GitHubPRTrigger trigger = JobInfoHelpers.triggerFrom(run.getParent(), GitHubPRTrigger.class);
         if (trigger == null) {
-            return true;
+            // silently skip. TODO implement error handler, like in publishers
```

Comment:
Can you please show me an example of what you mean "like in the publishers". I am trying to convert the http-request plugin and I have the same question.